packages/core/src/ObjectContainer.js

```diff
@@ -504,9 +518,50 @@ export default class ObjectContainer {
       }
     }
 
+    if (this._isSpread(name) && entry?.sharedInstance.length) {
+      let spreadEntry = Entry.createFromEntry(entry);
+
+      spreadEntry.sharedInstance = entry.sharedInstance.map(sharedInstance =>
+        this.get(sharedInstance)
+      );
+
+      this._entries.set(name, spreadEntry);
+      return spreadEntry;
+    }
+
     return entry;
   }
 
+  /**
+   * Checks whether the name is marked as optional.
+   *
+   * @param {string} name Name of a constant or alias,
+   *        factory function, class or interface constructor, or a fully
+   *        qualified namespace path.
+   * @return {boolean}
+   */
+  _isOptional(name) {
+    return (
+      (Array.isArray(name) && name?.[1]?.optional) ||
+      (typeof name === 'string' && name.includes('?'))
+    );
+  }
+
+  /**
+   * Checks whether the name is marked as spread.
+   *
+   * @param {string} name Name of a constant or alias,
+   *        factory function, class or interface constructor, or a fully
+   *        qualified namespace path.
+   * @return {boolean}
+   */
+  _isSpread(name) {
+    if (Array.isArray(name)) {
+      name = name?.[0];
```

Comment:
don't reassign the function param please, as it mutates the `arguments` object a is prone to errors.

```javascript
const normalizedName = Array.isArray(name) ? name[0] : name;

return typeof normalizedName === 'string' && normalizedName.startsWith('...');
```